src/app/space/create/deployments/apps/collapsible-deployment-info.component.spec.ts

```diff
@@ -0,0 +1,160 @@
+import { ChartModule } from 'patternfly-ng';
+import 'patternfly/dist/js/patternfly-settings.js';
+
+@Component({
+  selector: 'deployments-donut',
+  template: ''
+})
+class FakeDeploymentsDonutComponent {
+  @Input() mini: boolean;
+  @Input() spaceId: string;
+  @Input() applicationId: string;
+  @Input() environment: Environment;
+}
+
+@Component({
+  selector: 'deployment-graph-label',
+  template: ''
+})
+class FakeDeploymentGraphLabelComponent {
+  @Input() type: any;
+  @Input() dataMeasure: any;
+  @Input() value: any;
+  @Input() valueUpperBound: any;
+}
+
+@Component({
+  selector: 'pfng-chart-sparkline',
+  template: ''
+})
+class FakePfngChartSparkline {
+  @Input() config: any;
+  @Input() chartData: any;
+}
+
+describe('CollapsibleDeploymentInfoComponent', () => {
+  let component: CollapsibleDeploymentInfoComponent;
+  let fixture: ComponentFixture<CollapsibleDeploymentInfoComponent>;
+  let mockSvc: jasmine.SpyObj<DeploymentsService>;
+  let mockEnvironment: Environment;
+
+  beforeEach(() => {
+    mockSvc = createMock(DeploymentsService);
+    mockSvc.getVersion.and.returnValue(Observable.of('1.2.3'));
+    mockSvc.getCpuStat.and.returnValue(Observable.of({ used: 1, quota: 2 }));
+    mockSvc.getMemoryStat.and.returnValue(Observable.of({ used: 3, quota: 4, units: 'GB' }));
+    mockSvc.getAppUrl.and.returnValue(Observable.of('mockAppUrl'));
+    mockSvc.getConsoleUrl.and.returnValue(Observable.of('mockConsoleUrl'));
+    mockSvc.getLogsUrl.and.returnValue(Observable.of('mockLogsUrl'));
+    mockSvc.deleteApplication.and.returnValue(Observable.of('mockDeletedMessage'));
+
+    mockEnvironment = { name: 'mockEnvironment' } as Environment;
+
+    TestBed.configureTestingModule({
+      imports: [CollapseModule.forRoot()],
+      declarations: [
+        CollapsibleDeploymentInfoComponent,
+        FakeDeploymentsDonutComponent,
+        FakeDeploymentGraphLabelComponent,
+        FakePfngChartSparkline
+      ],
+      providers: [
+        { provide: DeploymentsService, useValue: mockSvc }
+      ]
+    });
+
+    fixture = TestBed.createComponent(CollapsibleDeploymentInfoComponent);
+    component = fixture.componentInstance;
+
+    component.collapsed = false;
+    component.applicationId = 'mockAppId';
+    component.environment = mockEnvironment;
+    component.spaceId = 'mockSpaceId';
+
+    fixture.detectChanges();
+  });
+
+  it('should generate unique chartIds for each CollapsibleDeploymentInfoComponent instance', () => {
+    const COLLAPSIBLE_COMPONENT_COUNT = 5;
+    const CHARTS_PER_COLLAPSIBLE_COMPONENT = 2;
+    let uniqueIdSet = new Set();
+    times(COLLAPSIBLE_COMPONENT_COUNT, () => {
+      let collapsibleComponent = new CollapsibleDeploymentInfoComponent(mockSvc);
+      uniqueIdSet.add(collapsibleComponent.cpuConfig.chartId);
+      uniqueIdSet.add(collapsibleComponent.memConfig.chartId);
+    });
+    expect(uniqueIdSet.size).toBe(COLLAPSIBLE_COMPONENT_COUNT * CHARTS_PER_COLLAPSIBLE_COMPONENT);
+  });
```

Comment:
I'm not sure that testing this in this manner is really needed - it might be sufficient to just assert that the cpuConfig and memCofig have differing chartIds for a single CollapsibleDeploymentInfoComponent. Iterating lots of times to "prove" that there are no ID collisions seems simultaneously overkill and fragile/unreliable.